File: matiasmasso/Mat.NET/DTO/DTOProductStat.cs
using System;
using System.Collections.Generic;

namespace DTO
{
    public class DTOProductStat : DTOProduct
    {
        public List<DTOYearMonth> Items { get; set; }

        public DTOProductStat() : base()
        {
            Items = new List<DTOYearMonth>();
        }

        public static DTOProductStat Factory(DTOProduct oProduct, List<Exception> exs)
        {
            DTOProductStat retval = new DTOProductStat();
            DTOBaseGuid.CopyPropertyValues<DTOProduct>(oProduct, retval, exs);
            return retval;
        }
    }
}

File: matiasmasso/Mat.NET/DT2/DTOProductStat.vb
Public Class DTOProductStat
    Inherits DTOProduct
    Property items As List(Of DTOYearMonth)

    Public Sub New()
        MyBase.New
        _items = New List(Of DTOYearMonth)
    End Sub

    Shared Shadows Function Factory(oProduct As DTOProduct, exs As List(Of Exception)) As DTOProductStat
        Dim retval As New DTOProductStat
        DTOBaseGuid.CopyPropertyValues(Of DTOProduct)(oProduct, retval, exs)
        Return retval
    End Function
End Class
